frontends/web/src/routes/account/send/utxos.tsx

```diff
@@ -14,92 +15,72 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  allScriptTypes,
+  getUTXOs,
+  ScriptType,
+  TUTXO,
+} from '../../../api/account';
+import { syncdone } from '../../../api/subscribe-legacy';
 import A from '../../../components/anchor/anchor';
 import { Dialog } from '../../../components/dialog/dialog';
 import { Button, Checkbox } from '../../../components/forms';
-import { ExpandOpen } from '../../../components/icon/icon';
+import { ExpandOpen } from '../../../components/icon';
 import { FiatConversion } from '../../../components/rates/rates';
-import { translate, TranslateProps } from '../../../decorators/translate';
 import { getScriptName } from '../utils';
-import { unsubscribe, UnsubscribeList } from '../../../utils/subscriptions';
-import { syncdone } from '../../../api/subscribe-legacy';
 import style from './utxos.module.css';
 
-interface UTXOsProps {
-    accountCode: string;
-    active: boolean;
-    explorerURL: string;
-    onChange: (selectedUTXO: SelectedUTXO) => void;
-    onClose: () => void;
-    // eslint-disable-next-line react/no-unused-prop-types
-    ref?: React.RefObject<any> // WithTranslation doesn't add ref prop correctly
-}
-
-export interface SelectedUTXO {
-    [key: string]: boolean;
-}
-
-export type Props = UTXOsProps & TranslateProps;
-
-interface State {
-    utxos: accountApi.UTXO[];
-    selectedUTXOs: SelectedUTXO;
-}
-
-export class UTXOsClass extends Component<Props, State> {
-  public readonly state: State = {
-    utxos: [],
-    selectedUTXOs: {},
-  };
-
-  private subscriptions: UnsubscribeList = [];
-
-  public componentDidMount() {
-    this.getUTXOs();
-    this.subscribe();
-  }
-
-  public componentWillUnmount() {
-    unsubscribe(this.subscriptions);
-  }
-
-  private subscribe() {
-    this.subscriptions.push(
-      syncdone(this.props.accountCode, () => this.getUTXOs()),
-    );
-  }
-
-  private getUTXOs() {
-    accountApi.getUTXOs(this.props.accountCode).then(utxos => {
-      this.setState({ utxos });
+export type TSelectedUTXOs = {
+  [key: string]: boolean;
+};
+
+type Props = {
+  accountCode: string;
+  active: boolean;
+  explorerURL: string;
+  onChange: (selectedUTXO: TSelectedUTXOs) => void;
+  onClose: () => void;
+};
+
+export const UTXOs = ({
+  accountCode,
+  active,
+  explorerURL,
+  onChange,
+  onClose,
+}: Props) => {
+  const { t } = useTranslation();
+  const [utxos, setUtxos] = useState<TUTXO[]>([]);
+  const [selectedUTXOs, setSelectedUTXOs] = useState<TSelectedUTXOs>({});
+
+  useEffect(() => {
+    getUTXOs(accountCode).then(setUtxos);
+    return () => setUtxos([]);
+  }, [accountCode]);
+
+  useEffect(() => {
+    const unsubscribe = syncdone(accountCode, () => {
+      getUTXOs(accountCode).then(setUtxos);
     });
-  }
-
-  public clear = () => {
-    this.setState({ selectedUTXOs: {} }, () => {
-      this.props.onChange(this.state.selectedUTXOs);
-    });
-  };
```

Comment:
this was so that the parent could clear this component, which then emptied the state. selectedUTXOs, and send it back to the parent.

changed to just clear the selectedUTXOs on the parent directly.